app/controllers/spree/subscriptions_controller.rb

```diff
@@ -23,24 +23,46 @@ def update
     end
 
     def cancel
-      respond_to do |format|
+      if @subscription.stripe_subscription_id.present?
         if @subscription.cancel
-          format.json { render json: {
-              subscription_id: @subscription.id,
-              flash: t(".success"),
-              method: Spree::Subscription::ACTION_REPRESENTATIONS[:cancel].upcase
-            }, status: 200
-          }
-          format.html { redirect_to account_path, success: t(".success") }
+          @stripe_subscription = Stripe::Subscription.retrieve(@subscription.stripe_subscription_id)
```

Comment:
what is subscription is not found?

Reply:
its the case for example stripe subscription is not assigned to our spree subscriptions for any error, then it should give error here and redirect to account page instead of code break
i have linked product subscription cancel with stripe subscription cancellation.